File: server-go/internal/dto/product.go
package dto

// request/response structs, replaces Pydantic schemas
type ProductResponse struct {
	Code        string `json:"code"`
	Eng         string `json:"eng"`
	Viet        string `json:"viet"`
	Alternative string `json:"alternative"`
	Brand       string `json:"brand"`
}

type ImageResponse struct {
	Code  	string `json:"code"`
	Images []string `json:"images"`
}

type MartinReportListResponse struct {
	Name string `json:"name"`
}

type MartinReportProductResponse struct {
	RowNo       int    `json:"row_no"`
	Code        string `json:"code"`
	Eng 			string `json:"eng"`
	Image      	*string `json:"image"`
	Quantity    int    `json:"quantity"`
}

File: client/src/types/product.ts

/** Matches server/internal/dto.ProductResponse. */
export interface ProductResponse {
    code: string;
    viet: string;
    eng: string;
    alternative: string;
    brand: string;
}

/** Matches server/internal/dto.ImageResponse. */
export interface ImageResponse {
    code: string;
    images: string[];
}

/** Matches server/internal/dto.MartinReportListResponse. */
export interface MartinReportListResponse {
    id: number;
    name: string;
}

/** Matches server/internal/dto.MartinReportProductResponse. */
export interface MartinReportProductResponse {
    row_no:      number;
    code:        string;
    description: string;
    image:       string | null;
    quantity:    number;
}

/** Request payload for creating a Martin report. */
export interface CreateMartinReportRequest {
    name: string;
}

/** Request payload for replacing the instrument rows in a Martin report. */
export interface UpdateMartinReportRequest {
    products: MartinReportProductResponse[];
}

/** Matches the error payload written by the server handlers. */
export interface ApiErrorResponse {
    error: string;
}
